### ISSUE TITLE
[Card product] - `aria-labelledby` referencing element that sometimes doesnt exist 

### ISSUE THREAD

**Issue by sofiamatulis** [Shopify/dawn#2097]
**Describe the current behavior**

In the card product.liquid file - `aria-labelledby` is referencing element that sometimes doesnt exist.

https://github.com/Shopify/dawn/commit/b10202cf32cb930607de4b0611cdde8d2bb24b29#diff-167fc70beb2d8c5dbf3e81a213fd3808f6b2e1d1c60590a6863724dc0dca3ea6R147

It references 2 elements: **CardLink-{{ section_id }}-{{ card_product.id }}**  which we have [here](https://github.com/Shopify/dawn/commit/b10202cf32cb930607de4b0611cdde8d2bb24b29#diff-167fc70beb2d8c5dbf3e81a213fd3808f6b2e1d1c60590a6863724dc0dca3ea6R145) and **Badge-{{ section_id }}-{{ card_product.id }}**  which we have [here](https://github.com/Shopify/dawn/commit/b10202cf32cb930607de4b0611cdde8d2bb24b29#diff-167fc70beb2d8c5dbf3e81a213fd3808f6b2e1d1c60590a6863724dc0dca3ea6R300)

It looks like the 2nd one is under an if statement which sometimes may not be true https://screenshot.click/03-37-zv2j7-6znsm.png so it doesnt render.

**Describe the expected behavior**
There should be an if statement in the aria-labelledby to only render when the element exists
